### ISSUE TITLE
default_port() takes a scheme but not a protocol()

### ISSUE THREAD

**Issue by AlexandreBossard** [cpp-netlib/url#135]
```cpp
auto const myurl = url("https://exampe.com");
url::default_port(myurl.protocol()).has_value(); // false
```

So, `protocol()` returns `https:` but `default_port()` needs a scheme (no ':'). Plus, there is no `url::scheme()` provided. Is this some overlooking ? Having to strip the ':' every time is annoying.

**Comment by glynos** [comment]
That's a simple fix, I'd appreciate a short review of #136 to make sure I captured what you asked for.

**Comment by glynos** [comment]
Fixed in #136
